packages/mobile/src/account/LinkBankAccountScreen.tsx

```diff
@@ -22,92 +26,167 @@ import VerificationPending from 'src/icons/VerificationPending'
 import { linkBankAccountStepTwoEnabledSelector } from '../app/selectors'
+import { fetchFinclusiveKyc } from './actions'
 import openPlaid, { handleOnEvent } from './openPlaid'
 
-interface StepOneProps {
-  kycStatus: KycStatus | undefined
-}
-
 function LinkBankAccountScreen() {
-  // Log a cancel event on a "back" action (hardware back button, swipe, or normal navigate back)
   const navigation = useNavigation()
-  React.useEffect(() => {
+
+  // Log a cancel event on a "back" action (hardware back button, swipe, or normal navigate back)
+  useEffect(() => {
     const unsubscribe = navigation.addListener('beforeRemove', (e) => {
       ValoraAnalytics.track(CICOEvents.link_bank_account_cancel)
     })
     // Unsubscribe will be called on unmount
     return unsubscribe
   }, [])
 
-  const kycStatus = useSelector(kycStatusSelector)
-  const stepTwoEnabled = useSelector(linkBankAccountStepTwoEnabledSelector)
-
   return (
     <SafeAreaView style={styles.body}>
       <ScrollView
         contentContainerStyle={styles.contentContainer}
         showsVerticalScrollIndicator={false}
       >
-        <StepOne kycStatus={kycStatus} />
-        <StepTwo disabled={!stepTwoEnabled || kycStatus !== KycStatus.Approved} />
+        <StepOne />
+        <StepTwo />
       </ScrollView>
     </SafeAreaView>
   )
 }
 
-export function StepOne({ kycStatus }: StepOneProps) {
+interface StepOneUIStateParams {
+  kycStatus: KycStatus | undefined
+  finclusiveKycStatus: FinclusiveKycStatus
+  errorFromPersona: boolean
+  successFromPersona: boolean
+  isInPersonaFlow: boolean
+}
+
+enum StepOneUIState {
+  Begin = 'Begin',
+  Failure = 'Failure',
+  Pending = 'Pending',
+  Completed = 'Completed',
+  Spinner = 'Spinner',
+}
+export function stepOneUIState({
+  kycStatus,
+  finclusiveKycStatus,
+  errorFromPersona,
+  successFromPersona,
+  isInPersonaFlow,
+}: StepOneUIStateParams): StepOneUIState {
+  // The persona onSuccess / onError hooks fire slightly after the user is bounced back
+  // to the LinkedBankAccount Screen. We show a spinner at this time to avoid a more
+  // unpleasent screen flash as the kyc states change
+  if (isInPersonaFlow) {
+    return StepOneUIState.Spinner
+  }
+
+  // Show the Completed Screen
+  if (finclusiveKycStatus === FinclusiveKycStatus.Accepted) {
+    return StepOneUIState.Completed
+  }
+
+  // Show the Pending Screen
+  const userCompletedPersona: (KycStatus | undefined)[] = [
+    KycStatus.Completed,
+    KycStatus.Approved,
+    KycStatus.NeedsReview,
+  ]
+  const finclusiveNotCompleted = [FinclusiveKycStatus.Submitted, FinclusiveKycStatus.InReview]
+  if (
+    successFromPersona ||
+    userCompletedPersona.includes(kycStatus) ||
+    finclusiveNotCompleted.includes(finclusiveKycStatus)
+  ) {
+    return StepOneUIState.Pending
+  }
+
+  // Show the Failed Screen
+  const personaFailed: (KycStatus | undefined)[] = [KycStatus.Failed, KycStatus.Declined]
+  if (
+    errorFromPersona ||
```

Comment:
I feel like that if onError hook is triggered, we prob want user to retry instead of just showing a failure state. Say if there's an internal application failure during the persona inquiry flow, we prob dont want to just show a failure screen

Reply:
The failure screen does have a retry button. The copy is not the best, but I think it works well enough for this scenario 
<img width="209" alt="Screen Shot 2022-02-28 at 11 11 25 AM" src="https://user-images.githubusercontent.com/8432644/156044009-5b866812-d679-43b6-8d17-9ac83b65a2e2.png">